nucleus/deployment/common/src/main/java/org/glassfish/deployment/common/GenericAnnotationDetector.java

```diff
@@ -151,8 +152,7 @@ public void scanArchive(ReadableArchive archive) {
                     } finally {
                         is.close();
                     }
-                } else if (entryName.endsWith(".jar") && 
-                    entryName.indexOf('/') == -1) {
+                } else if (!entryName.contains("/")) {
```

Comment:
why has this changed? It looks like it may produce side effects.

Reply:
Side effect when calling jarSubArchive.entries().

com.sun.enterprise.deployment.deploy.shared.InputJarArchive#entries(boolean) does not throw an IOException but a RuntimeException. The effect is the exception is not catch by the inner try-catch bloc but the outer one, skipping all other entries in the archive.